File: 1232. Check If It Is a Straight Line.py
class Solution:
    def checkStraightLine(self, coordinates: List[List[int]]) -> bool:
        # 思路：用两个坐标建立方程（求出斜率再带回去求出weight），然后其他的坐标进行验证
        # 另一种思路：对于任意三点而言，两两斜率均相等（使用乘法来避免除以0）
        (x0, y0), (x1, y1) = coordinates[: 2] # [:2] 返回0和1,不包含2
        for x, y in coordinates:
            if (x1 - x0) * (y - y1) != (x - x1) * (y1 - y0):
                return False
        return True

        

File: 1232. Check If It Is a Straight Line.java
class Solution {
    // 同python版思路，遍历元素，每三个元素之间两两的斜率均相同
    // int[][] 代表2d list
    public boolean checkStraightLine(int[][] coordinates) {
        // 这种创建变量的表达方式
        int x0 = coordinates[0][0], y0 = coordinates[0][1],
            x1 = coordinates[1][0], y1 = coordinates[1][1];
        int dx = x1 - x0, dy = y1 - y0;
        for (int[] co : coordinates) {
            int x = co[0], y = co[1];
            if (dx * (y - y1) != dy * (x - x1))
                return false;
        }
        
        return true;
    }
}
